mesonbuild/modules/windows.py

```diff
@@ -151,10 +156,20 @@ def add_target(src):
                 'depends': wrc_depends,
             }
 
-            # instruct binutils windres to generate a preprocessor depfile
             if rescomp_type == ResourceCompilerType.windres:
+                # instruct binutils windres to generate a preprocessor depfile
                 res_kwargs['depfile'] = res_kwargs['output'] + '.d'
                 res_kwargs['command'] += ['--preprocessor-arg=-MD', '--preprocessor-arg=-MQ@OUTPUT@', '--preprocessor-arg=-MF@DEPFILE@']
+            elif rescomp_type == ResourceCompilerType.wrc:
+                # instruct wrc to use the standard preprocessor in order to generate a depfile
+                comp = self.detect_compiler(state.environment.coredata.compilers[MachineChoice.HOST])
+                commands = comp.get_exelist() + comp.get_preprocess_only_args()
+                commands.append('-MD')
+                commands.append('-MQ@OUTPUT@')
+                commands.append('-MF@DEPFILE@')
+                res_kwargs['depfile'] = res_kwargs['output'] + '.d'
+                res_kwargs['command'] += ['--preprocessor', join_args(commands)]
```

Comment:
Do the expected -I flags, and -DRC_INVOKED get added to this command, or is it used literally?